lib/fluent/plugin/out_splunk_hec.rb

```diff
@@ -279,7 +279,9 @@ def format_metric(tag, time, record)
     end
 
     def construct_api
-      URI("#{@protocol}://#{@hec_host}:#{@hec_port}/services/collector")
+      host = @hec_host.delete_prefix("/").delete_suffix("/").split("/", 2)
+      host_suffix = host[1].nil? ? "" : "/#{host[1]}"
+      URI("https://#{host[0]}:#{@hec_port}#{host_suffix}/services/collector")
```

Comment:
it can only support one path, right? can it be updated so that it can support any number of it? 
like `https://mydomain.com/splunk/hec`

Reply:
maybe a better approach would be adding another parameter. `full_url`. 
use `full_url` if provided. if not build one using `host`, `port`, and `protocol` .
@chaitanyaphalak any thoughts?

Reply:
> it can only support one path, right? can it be updated so that it can support any number of it? like `https://mydomain.com/splunk/hec`

It will support any number of path as I am splitting host in 2 half only.
For `mydomain.com/splunk/hec`, it will split into `mydomain.com` and `splunk/hec` and will be join them with port value

Reply:
I like rock's idea

Reply:
@harshit-splunk i see. so it works. 
hmm still, this way of parsing is not intuitive for the user. 
instead of `full_url` we can add `url_path` parameter. 
default would be `""` but user can set it to `/splunk/hec` or anything.

However, having a `full_url` parameter is something that will benefit usability in the future. we can use it in the helm chart to reduce number of arguments that they need to pass. 
this way, we are consistent with olly chart. (ie `splunkPlatform.endpoint`)

Reply:
@harshit-splunk 
So I would like to suggest to edit this PR with `full_url` approach.